internal/cri/server/container_start_linux.go

```diff
@@ -31,11 +31,19 @@ func updateContainerIOOwner(ctx context.Context, cntr containerd.Container, conf
 		return nil, nil
 	}
 
-	// FIXME(fuweid): Ideally, the pipe owner should be aligned with process owner.
-	// No matter what user namespace container uses, it should work well. However,
-	// it breaks the sig-node conformance case - [when querying /stats/summary should report resource usage through the stats api].
+	// FIXME(fuweid):
+	//
+	// For builtin runc runtime, the pipe owner should be aligned with process
+	// owner. No matter what user namespace container uses, it should work
+	// well.
+	//
+	// However, gVisor runtime doesn't support runc.Options and no idea why
+	// adding options could breaks the sig-node conformance case [when querying /stats/summary should report resource usage through the stats api].
```

Comment:
For my understanding, would there be something that could (should?) also be addressed by the gVisor maintainers? (if so, did we raise it with them?)

Reply:
> also be addressed by the gVisor maintainers?

After check gVisor code (like https://github.com/google/gvisor/blob/b78f2ee7c4c393990a84298dd0f200e927b18dab/pkg/shim/runsc/service.go#L213), I think gVisor project should address this compatible issue, even if it's kind of painful. 

The main issue seems to be that containerd lacks abstract layer for configuring new task in shim. In containerd client SDK, new task always uses runc option by default. For the out-of-tree runtimes, it's unlikely to support them in CRI plugin if there is no common layer for them. It's weird to check runtime name, since it's not reliable. So, I use runc option only if pod enables pod-level user namespace. It's workaround.

Other option is that we can automatically adjust pipe owner. But it's plan B.

> if so, did we raise it with them?

The issue reporter seems not to be in the gVisor team. I can tag gVisor team in that issue https://github.com/containerd/containerd/issues/11091.